### PR TITLE
Configurable + Multiple application source

### PR DESCRIPTION
**Pull request by qparis**

### CODE REVIEW

**Comment by qparis** [comment]
@plata Can you look at this pull request?

File: phoenicis-tools/src/main/java/com/playonlinux/tools/http/Downloader.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ private void saveConnectionToStream(URL url,
                 totalDataRead += i;
                 bufferedOutputStream.write(data, 0, i);
 
-                percentage = totalDataRead * 100 / fileSize;
+                percentage = ((float) totalDataRead * 100) / ((float) fileSize);
```
Comment (plata) [inline phoenicis-tools/src/main/java/com/playonlinux/tools/http/Downloader.java]: Please don't mix unrelated commits in one pull request.

File: phoenicis-tools/src/main/java/com/playonlinux/tools/files/FileUtilities.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public void remove(File fileToDelete) throws IOException {
         if(fileToDelete.isDirectory()) {
             FileUtils.deleteDirectory(fileToDelete);
         } else {
-            fileToDelete.delete();
+            Files.deleteIfExists(fileToDelete.toPath());
```
Comment (plata) [inline phoenicis-tools/src/main/java/com/playonlinux/tools/files/FileUtilities.java]: Please don't mix unrelated commits in one pull request.

File: phoenicis-tools/src/main/java/com/playonlinux/tools/checksum/ChecksumCalculator.java [hunk 3]
```diff
@@ -67,7 +67,7 @@ public String calculate(File fileToCheck, String algorithm, Consumer<ProgressEnt
             messageDigest.update(bytes, 0, numBytes);
             readBytes += numBytes;
             if(sizeInBytes != 0L) {
-                double percentage = (double) readBytes / (double) sizeInBytes * (double) 100;
+                double percentage = (double) readBytes * 100. / (double) sizeInBytes;
```
Comment (plata) [inline phoenicis-tools/src/main/java/com/playonlinux/tools/checksum/ChecksumCalculator.java]: Please don't mix unrelated commits in one pull request.

File: phoenicis-apps/src/main/java/com/playonlinux/apps/ConfigurableApplicationSource.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    private final String sourceUrl;
+    private final LocalApplicationsSource.Factory localApplicationsSourceFactory;
+
+    public ConfigurableApplicationSource(String sourceUrl, LocalApplicationsSource.Factory localApplicationsSourceFactory) {
+        this.sourceUrl = sourceUrl;
+        this.localApplicationsSourceFactory = localApplicationsSourceFactory;
+    }
+
+    @Override
+    public List<CategoryDTO> fetchInstallableApplications() {
```
Comment (plata) [inline phoenicis-apps/src/main/java/com/playonlinux/apps/ConfigurableApplicationSource.java]: The MultipleApplicationSources will be created every time this is called. It would be better to create the MultipleApplicationSources only once in the constructor.

File: phoenicis-apps/src/main/java/com/playonlinux/apps/MultipleApplicationSource.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    MultipleApplicationSource(List<ApplicationsSource> applicationsSources) {
+        this.applicationSources = applicationsSources;
+    }
+
+    @Override
+    public List<CategoryDTO> fetchInstallableApplications() {
+        ApplicationsSource lastApplicationSource = new NullApplicationSource();
+
+        for (ApplicationsSource applicationSource : applicationSources) {
+            lastApplicationSource = new TeeApplicationsSource(lastApplicationSource, applicationSource);
```
Comment (plata) [inline phoenicis-apps/src/main/java/com/playonlinux/apps/MultipleApplicationSource.java]: The lastApplicationsSource will be created every time this is called. It would be better to create the lastApplicationsSource only once in the constructor.

File: phoenicis-javafx/src/main/java/com/playonlinux/javafx/controller/MainController.java [hunk 6]
```diff
@@ -75,18 +75,18 @@ public void show() {
     }
 
     public void setOnClose(Runnable onClose) {
-        this.mainWindow.setOnCloseRequest(event ->{
```
Comment (plata) [inline phoenicis-javafx/src/main/java/com/playonlinux/javafx/controller/MainController.java]: Please don't mix unrelated commits in one pull request.
